### DISCUSSION TITLE
Auto Split into Train/Val/Test Sets

### DISCUSSION THREAD

**Discussion by jquaglia-uss** [ultralytics/hub#436] [Ideas]
One of the things that makes training models more tedious and time consuming is dataset prep.  Mainly, resizing images and putting them into the proper set (Train/Val/Test).  A massively useful feature would be to have this step automatically done during upload.  

Current functionality is specifying a folder that already contains train/val/test sets.

Useful enhancement of the product would be the ability to just select a folder of labeled images (this is probably most useful for classification tasks) and the upload process could resize/assign splits for the user.

-JQ

**Comment by crapthings** [comment]
therer's a autosplit function in utils
